ts/session/onions/onionSend.ts

```diff
@@ -287,24 +292,35 @@ async function sendJsonViaOnionV4ToSogs(
     abortSignal,
     headers: includedHeaders,
     throwErrors,
+    includeAuthHeaders = true,
     timeoutMs,
   } = sendOptions;
+
   if (!endpoint.startsWith('/')) {
     throw new Error('endpoint needs a leading /');
   }
+
   const builtUrl = new URL(`${serverUrl}${endpoint}`);
-  let headersWithSogsHeadersIfNeeded = await OpenGroupPollingUtils.getOurOpenGroupHeaders(
-    serverPubkey,
-    endpoint,
-    method,
-    blinded,
-    stringifiedBody
-  );
+  let headersWithSogsHeadersIfNeeded: Record<string, any> = includedHeaders || {};
 
-  if (!headersWithSogsHeadersIfNeeded) {
-    return null;
+  if (includeAuthHeaders) {
+    const ourHeaders = await OpenGroupPollingUtils.getOurOpenGroupHeaders(
+      serverPubkey,
+      endpoint,
+      method,
+      blinded,
+      stringifiedBody
+    );
+
+    if (!ourHeaders) {
+      return null;
```

Comment:
Why the null return and not throw?

Reply:
I kept it the same way it works. if I make it throw, I need to add a try catch everywhere it is called.
the signature of the function already allows for null, so I assume all the current calls to this function are checking the result instead of try/catch,.